pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -299,3 +320,21 @@ func (m *smcpv2mutator) SetOpenShiftRouteEnabled(value bool) {
 
 	m.patches = append(m.patches, jsonpatch.NewPatch("add", "/spec/gateways", *gateways))
 }
+
+func (m *smcpv1mutator) IsTracingTypeSpecified() *bool {
+	return nil
+}
+
+func (m *smcpv1mutator) SetTracingType(value string) {}
+
+func (m *smcpv2mutator) IsTracingTypeSpecified() *bool {
+	tracingType := m.smcp.Spec.Tracing
+
+	if tracingType == nil {
+		return nil
+	
+}
```

Comment:
I think returning a `bool` makes more sense than returning `*bool` here. We only need to know whether this field has been explicitly set by a user (`true`) or not specified (`false`). The nil value isn't useful.

Also, you are not checking the `m.smcp.Spec.Tracing.Type` here.